modular/manager/bucket_migrate_scheduler.go

```diff
@@ -535,33 +614,66 @@ func (s *BucketMigrateScheduler) replaceExitingSP(secondarySPIDs []uint32) ([]ui
 	}
 
 	return replacedSPIDs, nil
-
 }
 
-func (s *BucketMigrateScheduler) generateBucketMigrateGVGExecuteUnitFromDB(primarySPGVGList []*virtualgrouptypes.GlobalVirtualGroup, bucketID uint64, executePlan *BucketMigrateExecutePlan) (*BucketMigrateExecutePlan, error) {
-	// TODO: check every migrateGVGUnitMeta must match primarySPGVGList
-	// notFinishedGVGList
+func (s *BucketMigrateScheduler) generateBucketMigrateGVGExecuteUnitFromDB(primarySPGVGList []*virtualgrouptypes.GlobalVirtualGroup, bucketID uint64, executePlan *BucketMigrateExecutePlan, event *storagetypes.EventMigrationBucket) (*BucketMigrateExecutePlan, error) {
+	// check every migrateGVGUnitMeta must match primarySPGVGList
+
+	// 1) check chain's bucket status
+	expected, err := s.checkBucketFromChain(bucketID, storagetypes.BUCKET_STATUS_MIGRATING)
+	if err != nil {
+		return nil, err
+	}
+	if !expected {
+		return nil, nil
+	}
+
+	chainGvgMaps := make(map[uint32]*virtualgrouptypes.GlobalVirtualGroup)
```

Comment:
If the map size is known, you can directly make the memory of the corresponding size to improve performance

```
chainGvgMaps := make(map[uint32]*virtualgrouptypes.GlobalVirtualGroup, len(primarySPGVGList))
```